### DISCUSSION TITLE
How to combine nested writes with raw queries

### DISCUSSION THREAD

**Discussion by benknab** [prisma/prisma#5804] [Q&A]
I have a schema where one of my models has an unsupported PostGIS polygon type:

```prisma
enum Crop {
  Wheat
  Corn
  Sunflower
}

model Cultivation {
  id      Int       @id @default(autoincrement())
  crop    Crop
  start   DateTime
  end     DateTime?
  fieldId Int

  field Field @relation(fields: [fieldId], references: [id])
}

model Field {
  id           Int @id @default(autoincrement())
  name         String
  cultivations Cultivation[]

  // Unsupported GIS type
  // area: Polygon
}
```

This is my current code to create a field with a cultivation:
```js
const field = await prisma.$queryRaw`...`
const cultivation = await prisma.cultivation.create({ data: { fieldId: field.id, ... } })
```

I would like to ensure transactionality on the database level but I'm not sure how. I can't use nested writes with `prisma.cultivation.create` because I can't set the field's polygon field and I can't wrap them in `prisma.$transaction` because I need the field's id to create the cultivation.

Is there a way to somehow combine dependent writes with raw queries? 
As a last resort, I could create the cultivation in the same raw query as the field but I'm hoping for something simpler.

Thanks!

**Comment by ryands17** [comment]
@bkiac 👋 

I don't think this is possible right now as you cannot use the field from the 1st query into another with [Prisma's transaction flow](https://www.prisma.io/blog/how-prisma-supports-transactions-x45s1d5l0ww1) so you would need to use a single raw query for now in this case.

This should improve once this [request](https://github.com/prisma/prisma/issues/5683) is done so I would suggest adding a 👍 along with your use case as well :)

**Reply by benknab** [reply]
Thanks for the quick reply, I linked this discussion in the request!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
